### PR TITLE
Upgrade to Conan 2

### LINKED ISSUE #54: Update to Conan 2

### PR DESCRIPTION
**Pull request by kyllingstad**
Fixes #54.

This depends on open-simulation-platform/libcosim#747. (Once that PR has been merged, the version specification of the `libcosim` requirement should be changed so it no longer points to the `osp/testing-feature...` branch.)

I've added `shared` and `fPIC` options, which seem to be rather standard for library packages these days. (Previously, we only built the shared library version.)

### CODE REVIEW

**Comment by kyllingstad** [comment]
I got some build errors related to Boost.Fibers that I couldn't quite figure out, so rather than try to fix them, I simply removed the explicit dependency on Boost. Libcosim no longer uses Boost.Fibers, and everything we used it for here could be done with stdlib functionality from `<future>`, so I think this is a net win.

Note that there is still an *implicit* dependency on Boost, and this PR depends on open-simulation-platform/libcosim#753, which should be merged first.

File: conanfile.py [hunk 1]
```diff
@@ -1,47 +1,87 @@
...
+
+    # Dependencies/requirements
+    tool_requires = (
+        "cmake/[>=3.15]",
+        "doxygen/[>=1.8]",
+    )
     requires = (
-        "libcosim/0.10.2@osp/stable"
-        )
+        "libcosim/0.10.3@osp/testing-bugfix_transitive-libs-boost",
```
Comment (eidekrist) [inline conanfile.py:48]: Maybe this have been changed to `libcosim/0.10.3@osp/stable` before merging.

Comment (kyllingstad) [reply conanfile.py:48]: Oops! I've pushed a hotfix to `master` now.

Comment (kyllingstad) [reply conanfile.py:48]: Looks like my hotfix broke master. :( I guess the issue is that the upstream fix in open-simulation-platform/libcosim#753 is in libcosim `master`, but not in any release yet, so it's not in the `stable` channel. I don't know what the right way to fix this is. Revert all changes to libcosimc `master` and reopen this PR again, or make libcosimc `master` depend on the libcosim `testing` channel which tracks the `master` branch?

**Pull request merged**